lib/src/README.md

```diff
@@ -1,41 +1,64 @@
-For each header file, we create a translation unit and parse the `declarations` in it
-to the bindings.
-2. User can provide `header filters` to select which declaration from a particular header file should be added to the generated bindings. The can provide a list of header 'names' to include/exclude.
-We compare the header file name (not the exact path) to decide.
-The default behaviour is to include everything that's included when parsing a header.
-3. Use can provide Compiler options, which are passed to clang compiler as it is.
-4. All bindings are generated in a single file.
+The libclang wrapper can be _manually_ generated using `pub run ffigen:setup`. See [setup.dart](#setup.dart) for details.
 
-# Code Details
-## Modules
-We are using libclang to parse header files.
-This project is roughly divided in 3 major modules -
-### code_generator
-Converts a library(all bindings) to an actual string representation.
-- Library (output of the header parser).
-- Writer (provides configurations for generating bindings).
-- Binding (base class for all bindings - Func, Struc, Global, EnumClass, etc).
-### config_provider
-This takes care of validating user config files, printing config warnings and errors,
-converting config.yaml to a format the header_parser can use.
-- Spec (represents a single config, which a user can provide in the config file).
-- Config (holds all the config which will be required by header parser).
-### header_parser
-Uses libclang to convert the header to a Library which is then used by code_generator.
-- clang_bindings (bindings to libclang which are used for parsing).
-- sub_parsers (each sub-parser parses a particular kind of declaration - struct, function, typedef, enum).
-- type_extractor (extracts types from variables, function parameters, return types).
-- includer (tells what should/shouldn't be included depending of config).
-- parser (Main Entrypoint) (creates translation units for all header files, and sets up parsing them).
-- translation_unit_parser (parses header files, splits declarations and feeds them to their respective sub_parsers).
+The generated file is placed in the project's `.dart_tool/ffigen` folder, the file name also specifies the ffigen version (E.g - `_v0_2_4_libclang_wrapper.dylib`), this helps ensure the correct wrapper is being used for its corresponding version.
+
+This dynamic libary is then used by [Header Parser](#header-parser) for parsing C files.
+## Bindings
+The config file for generating bindings is `tool/libclang_config.yaml`. The bindings are generated to `lib/src/header_parser/clang_bindings/clang_bindings.dart`. These are used by [Header Parser](#header-parser) for calling libclang functions.
+# Scripts
+## ffigen.dart
+This is the main entry point for the user-  `pub run ffigen`
+- Command line options
+    - `--verbose`: set log level
+    - `--config`: for specifying a config file.
+- Checks if dynamic library is created and is up to date. If not, it tries to auto-create it.
```

Comment:
All the following sentences miss a subject (which is implied to be "The script" or `ffigen.dart`). Maybe rewrite the sentences to include the subject so that they are grammatically correct.

(And the rest of the file.)